src/core/etl/src/Flow/ETL/Function/HTMLQuerySelector.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Function;
+
+use function Flow\Types\DSL\type_html;
+use Dom\HTMLDocument;
+use Flow\ETL\Row;
+
+final class HTMLQuerySelector extends ScalarFunctionChain
+{
+    public function __construct(
+        private readonly mixed $value,
+        private readonly ScalarFunction|string $path,
+    ) {
+    }
+
+    public function eval(Row $row) : mixed
+    {
+        $value = (new Parameter($this->value))->as($row, type_html());
+        $path = (new Parameter($this->path))->asString($row);
+
+        if (null === $value || null === $path) {
+            return null;
+        }
+
+        if (!\class_exists('\Dom\HTMLDocument')) {
```

Comment:
I think it would be safer to simply throw an exception here since it's not even about the mistake and fault tolerance, without this class query selector can't be even used, which means this function works only on PHP 8.4+